A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,432 @@
+}
+```
+
+The EDS policy will use the XdsClient object passed in from the xds
+resolver to query for the `ClusterLoadAssignment` resource with the
+EDS service name from the LB policy configuration.
+
+Note that unlike the old grpclb policy, which just received a flat list
+of backend servers from the balancer to round-robin across, the EDS policy 
+receives a hierarchical list, where the servers are grouped into
+localities, and the localities are grouped into priorities.  The EDS
+policy will be responsible for both selecting the highest priority set
+of localities that are reachable and distributing the traffic across the
+localities in the selected priority based on the locality weights.
+
+Note that the EDS policy will *not* support
+[overprovisioning](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/overprovisioning),
+which is different from Envoy.  Envoy takes the
+overprovisioning into account in both [locality-weighted load
+balancing](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/locality_weight)
+and [priority
+fail-over](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/priority),
+but gRPC assumes that the xDS server will update it to redirect traffic
+when this kind of graceful fail-over is needed.
+
+Unlike the grpclb policy, which relied on server-side load reporting to
+the balancer, the EDS policy will perform client-side load reporting
+using the [LRS
+protocol](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/load_stats/v2/lrs.proto).
+Note that the EDS policy will not support per-endpoint stats; it will
+report only per-locality stats.  Currently, the EDS policy will report
+only client-side stats; in the future, we will support also reporting
+backend server stats reported to the client via
+[ORCA](https://docs.google.com/document/d/1NSnK3346BkBo1JUU3I9I5NYYnaJZQPt8_Z_XCBCI3uA/edit).
+
+### How gRPC Will Use the xDS API
+
+gRPC does not have exactly the same set of features as Envoy, so there
+are by necessity some small differences between how the two clients use
+the xDS API.  This section documents exactly what gRPC requires from the
+xDS server and what xDS features it does and does not support.
+
+Because there are many fields in the xDS APIs that gRPC will not use in
+its initial version, the gRPC client must in general ignore any fields
+it does not support.  This does mean that if we add support for one of
+these fields in a future version and users enable use of the feature in
+the management server, older clients will continue to ignore the field,
+so they may not actually get the new behavior.  However, the alternative
+is even worse: if we make the client strict (i.e., it will fail if the
+management server populates fields that the client does not yet support),
+then the same scenario will cause the older clients to fail instead of
+just ignoring the new feature.
```

Comment:
Should we be using client capabilities more aggressively to avoid the problem of unknown client feature support?